File: morethannDk.py
def ndk(arr,k):
    dic  = {}
    for num in arr:
        try:
            dic[num] = dic[num] + 1
        except:
            dic[num] = 1
    
    div = len(arr)//k
    l = []
    for d in dic:
        if dic[d] >= div:
            l.append(d)
    return l

if __name__ == "__main__":
    arr = [4,5,6,7,8,4,4]
    k = 3
    l = ndk(arr,k)
    print(l)

File: MoreThanNdK.java
import java.util.*;

public class MoreThanNdK{
    public static void ndk(int[] arr,int n,int k)
    {
        Dictionary d = new HashTable();

        for(int i : arr)
        {
            try {
                d.put(i,d.get(i) + 1);
            } catch (Exception e) {
                d.put(i,1);
            }
        }

        for(Enumeration e = d.keys() ; d.hasMoreElements();){
            if(e.nextElement() > (Integer)n/k){
                System.out.print(d.keys()+" ");
            }
        }
    }
    public static void main(String[] args) {
        int[] arr = {4,5,6,7,8,4,4};
        int k = 3;
        int n = arr.length;

        ndk(arr,n,k);
        
    }
}